pkg/ring/ring_test.go

```diff
@@ -132,6 +132,62 @@ func TestAddIngesterReplacesExistingTokens(t *testing.T) {
 	require.Equal(t, newTokens, r.Ingesters[ing1Name].Tokens)
 }
 
+func TestRing_Get_ZoneAwarenessWithIngesterLeaving(t *testing.T) {
```

Comment:
Could you also add a test to cover the case you're running a number of zones > replication factor, please? I would like to make sure it behaves correctly too.